packages/core/src/baseclient.ts

```diff
@@ -280,7 +276,7 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
       return scope.applyToEvent(prepared, hint, Math.min(maxBreadcrumbs, MAX_BREADCRUMBS));
     }
 
-    return prepared;
+    return SyncPromise.resolve(prepared as SentryEvent | null);
```

Comment:
I improved the syntax but this has to be done otherwise we get a build/lint error 
```
Error: ../core/src/baseclient.ts:278:5 - error TS2322: Type 'SyncPromise<SentryEvent>' is not assignable to type 'SyncPromise<SentryEvent | null>'.
  Types of property 'handlers' are incompatible.
    Type 'Handler<SentryEvent, any>[]' is not assignable to type 'Handler<SentryEvent | null, any>[]'.
      Type 'Handler<SentryEvent, any>' is not assignable to type 'Handler<SentryEvent | null, any>'.
        Type 'SentryEvent | null' is not assignable to type 'SentryEvent'.
          Type 'null' is not assignable to type 'SentryEvent'.

278     return SyncPromise.resolve(prepared);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```